BrickKit/bricks/src/androidTest/java/com/wayfair/brickkit/BrickRecyclerItemDecorationTest.kt

```diff
@@ -0,0 +1,132 @@
+        whenever(brickPadding.innerBottomPadding).thenReturn(INNER_BOTTOM)
+        whenever(brickPadding.outerLeftPadding).thenReturn(OUTER_LEFT)
+        whenever(brickPadding.outerTopPadding).thenReturn(OUTER_TOP)
+        whenever(brickPadding.outerRightPadding).thenReturn(OUTER_RIGHT)
+        whenever(brickPadding.outerBottomPadding).thenReturn(OUTER_BOTTOM)
+        whenever(brick.padding).thenReturn(brickPadding)
+        whenever(parent.getChildAdapterPosition(view)).thenReturn(0)
+        bricks.add(brick)
+    }
+
+    @Test
+    fun testInvalidNullItem() {
+        bricks.clear()
+
+        itemDecoration.getItemOffsets(outRect, view, parent, state)
+
+        assertEquals(0, outRect.left.toLong())
+        assertEquals(0, outRect.top.toLong())
+        assertEquals(0, outRect.right.toLong())
+        assertEquals(0, outRect.bottom.toLong())
+    }
+
+    @Test
+    fun testInvalidWrongDataManager() {
+        val adapter = mock<BrickRecyclerAdapter>()
+        whenever(parent.adapter).thenReturn(adapter)
+        whenever(adapter.brickDataManager).thenReturn(mock())
+
+        itemDecoration.getItemOffsets(outRect, view, parent, state)
+
+        assertEquals(0, outRect.left.toLong())
+        assertEquals(0, outRect.top.toLong())
+        assertEquals(0, outRect.right.toLong())
+        assertEquals(0, outRect.bottom.toLong())
+    }
+
+    @Test
+    fun testInvalidWrongAdapterType() {
+        whenever(parent.adapter).thenReturn(mock())
+
+        itemDecoration.getItemOffsets(outRect, view, parent, state)
+
+        assertEquals(0, outRect.left.toLong())
+        assertEquals(0, outRect.top.toLong())
+        assertEquals(0, outRect.right.toLong())
+        assertEquals(0, outRect.bottom.toLong())
+    }
+
+    @Test
+    fun testAllOuter() {
+        whenever(brick.isInFirstRow).thenReturn(true)
+        whenever(brick.isInLastRow).thenReturn(true)
+        whenever(brick.isOnLeftWall).thenReturn(true)
+        whenever(brick.isOnRightWall).thenReturn(true)
+
+        itemDecoration.getItemOffsets(outRect, view, parent, state)
+
+        assertEquals(OUTER_LEFT.toLong(), outRect.left.toLong())
+        assertEquals(OUTER_TOP.toLong(), outRect.top.toLong())
+        assertEquals(OUTER_RIGHT.toLong(), outRect.right.toLong())
+        assertEquals(OUTER_BOTTOM.toLong(), outRect.bottom.toLong())
+    }
+
+    @Test
+    fun testAllInner() {
+        whenever(brick.isInFirstRow).thenReturn(false)
+        whenever(brick.isInLastRow).thenReturn(false)
+        whenever(brick.isOnLeftWall).thenReturn(false)
+        whenever(brick.isOnRightWall).thenReturn(false)
+
+        itemDecoration.getItemOffsets(outRect, view, parent, state)
+
+        assertEquals(INNER_LEFT.toLong(), outRect.left.toLong())
+        assertEquals(INNER_TOP.toLong(), outRect.top.toLong())
+        assertEquals(INNER_RIGHT.toLong(), outRect.right.toLong())
+        assertEquals(INNER_BOTTOM.toLong(), outRect.bottom.toLong())
+    }
+
+    companion object {
```

Comment:
for tests, it's usually easier to make the companion private since most stuff is private

Reply:
I think I started doing this looking at the generated code once (private companion vs private const produced different outputs)